src/win32/ComHeapPtr.hxx

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2020 The Music Player Daemon Project
+ * http://www.musicpd.org
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#ifndef MPD_WIN32_COMHEAPPTR_HXX
+#define MPD_WIN32_COMHEAPPTR_HXX
+
+#include <objbase.h>
+#include <utility>
+#include <windows.h>
+
+template <typename T>
+class ComHeapPtr {
+public:
+	ComHeapPtr() noexcept = default;
+	ComHeapPtr(std::nullptr_t) noexcept {}
+	ComHeapPtr(const ComHeapPtr &rhs) noexcept = delete;
```

Comment:
If a move constructor exists, then this line is redundant. You can state that explicitly and keep it if you prefer to - I usually prefer to not to, so it boils down to personal taste.